source/dubregistry/registry.d

```diff
@@ -431,11 +431,21 @@ class DubRegistry {
 
 	void updatePackages()
 	{
+		import std.random : randomShuffle;
+
 		logDiagnostic("Triggering package update...");
 		// update stat distributions before score packages
 		m_statDistributions = m_db.getStatDistributions();
-		foreach (packname; this.availablePackages)
-			triggerPackageUpdate(packname);
+
+		// shuffle the package list to ensure a uniform probability of each
+		// package getting chosen over time, regardless of how long it takes
+		// to process the whole list and whether the process gets restarted
+		// in-between
+		auto allpacks = this.availablePackages.array;
+		randomShuffle(allpacks);
+		foreach (packname; allpacks)
```

Comment:
instead of allocating, you could use the randomCover function from std.random

Reply:
Turns out it still has to allocate, because `randomCover` requires a random access range.

Reply:
ah yeah makes sense, probably doesn't make sense to implement random access range on the cursor object because it would probably worsen performance.

I think `randomCover` also allocates a list of indices to iterate over so randomShuffle is probably the best option to use here. Maybe include a comment about this to avoid future code review thinking randomCover could be used instead?